src/realm/util/interprocess_condvar.cpp

```diff
@@ -208,8 +262,51 @@ void InterprocessCondVar::wait(InterprocessMutex& m, const struct timespec* tp)
     // postcondition: regardless of cause for return (timeout or notification),
     // the lock is held.
     REALM_ASSERT(m_shared_part);
+
 #ifdef REALM_CONDVAR_EMULATION
 
+#ifdef _WIN32
+    int64_t wait_milliseconds = INFINITE;
+
+    if (tp) {
+        timeval tp2;
+        gettimeofday(&tp2, nullptr);
+        wait_milliseconds = timediff(tp2, *tp);
+        if (wait_milliseconds < 0)
+            wait_milliseconds = 0;
+    }
+
+    m_shared_part->m_waiters_countlock.lock();
```

Comment:
yes. It's initialized like `shared_part.m_waiters_countlock.init_as_process_shared(true);`